api/pkg/webhooks/manager_test.go

```diff
@@ -0,0 +1,336 @@
+						},
+					},
+				},
+			},
+			eventList:     []EventType{"event1"},
+			expectedError: true,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			_, err := InitializeWebhooks(test.cfg, test.eventList)
+			if err != nil && !test.expectedError || err == nil && test.expectedError {
+				t.Errorf("expected error: %v, got: %v", test.expectedError, err)
+			}
+		})
+	}
+}
+func TestInvokeWebhooksSimple(t *testing.T) {
+	// Setup mock WebhookClient with specific behaviors
+	response := []byte(`{"data":"abc"}`)
+	mockClient := &MockWebhookClient{}
+	// Expect Invoke to return response from client and no error
+	mockClient.On("Invoke", mock.Anything, mock.Anything).Return(response, nil).Once()
+	mockClient.On("IsAsync").Return(false)
+	mockClient.On("IsFinalResponse").Return(true)
+	mockClient.On("GetUseDataFrom").Return("")
+	mockClient.On("GetName").Return("webhook1")
+
+	// Setup WebhookManager with the mock client
+	webhookManager := &SimpleWebhookManager{
+		WebhookClients: map[EventType][]WebhookClient{
+			"validEvent": {mockClient},
+		},
+	}
+
+	// Execution
+	err := webhookManager.InvokeWebhooks(
+		context.Background(),
+		"validEvent",
+		&testPayloadData,
+		onSuccess(t, response),
+		onError,
+	)
+
+	// Assertion
+	assert.NoError(t, err)           // Expect no error
+	mockClient.AssertExpectations(t) // Verify that expectations on the mock client were met
+}
+
+func TestInvokeMultipleSyncWebhooks(t *testing.T) {
+	// Setup mock WebhookClient with specific behaviors
+	webhook1Result := []byte(`{"result": "xyz"}`)
+	response := []byte(`{"data":"abc"}`)
+	mockClient := &MockWebhookClient{}
+	// Expect Invoke to return response from client and no error
+	mockClient.On("Invoke", mock.Anything, mock.Anything).Return(webhook1Result, nil)
+	mockClient.On("IsAsync").Return(false)
+	mockClient.On("GetName").Return("webhook1")
+	mockClient.On("IsFinalResponse").Return(false)
+	mockClient.On("GetUseDataFrom").Return("")
+
+	mockClient2 := &MockWebhookClient{}
+	mockClient2.On("Invoke", mock.Anything, mock.MatchedBy(func(payload []byte) bool {
+		if diff := cmp.Diff(webhook1Result, payload); diff != "" {
+			fmt.Println("diff", diff)
+			return false
+		}
+		return true
+	})).Return(response, nil)
+	mockClient2.On("IsAsync").Return(false)
+	mockClient2.On("GetName").Return("webhook2")
+	mockClient2.On("GetUseDataFrom").Return("webhook1")
+	mockClient2.On("IsFinalResponse").Return(true)
+	// Setup WebhookManager with the mock client
+	webhookManager := &SimpleWebhookManager{
+		WebhookClients: map[EventType][]WebhookClient{
+			"validEvent": {mockClient, mockClient2},
+		},
+	}
+	// Execution
+	err := webhookManager.InvokeWebhooks(
+		context.Background(),
+		"validEvent",
+		&testPayloadData,
+		onSuccess(t, response),
+		onError,
+	)
+
+	// Assertion
+	assert.NoError(t, err)
+	mockClient.AssertExpectations(t) // Verify that expectations on the mock client were met
+	// t.Errorf("test")
```

Comment:
Oh I missed this out; you probably want to remove this also 😅

Reply:
fixed